noggin/controller/authentication.py

```diff
@@ -57,3 +59,40 @@ def login():
                 )
 
     return render_template('login.html', login_form=form)
+
+
+@app.route('/otp/sync/', methods=['GET', 'POST'])
+def otp_sync():
+    form = SyncTokenForm()
+    if form.validate_on_submit():
+        try:
+            ipa = untouched_ipa_client(app)
+            response = ipa.otptoken_sync(
+                user=form.username.data,
+                password=form.password.data,
+                first_code=form.first_code.data,
+                second_code=form.second_code.data,
+                token=form.token.data,
+            )
+
+            if (
+                response.status_code == 200
+                and "Token sync rejected" not in response.text
+            ):
+                flash('Token successfully synchronized', category='success')
```

Comment:
There should be a redirect here instead of at the bottom, if we start using the form's errors.